edenai_apis/apis/microsoft/microsoft_helpers.py

```diff
@@ -321,306 +334,55 @@ def normalize_invoice_result(response):
-                .get("bounding_regions", [{}])[0]
-                .get("page_number")
-                == idx + 1
-                else None
-            )
-
-            invoice_number = (
-                fields.get("InvoiceId", default_dict).get("value")
-                if fields.get("InvoiceId", {})
-                .get("bounding_regions", [{}])[0]
-                .get("page_number")
-                == idx + 1
-                else None
-            )
-            date = (
-                format_date(fields.get("InvoiceDate", default_dict).get("value"))
-                if fields.get("InvoiceDate", {})
-                .get("bounding_regions", [{}])[0]
-                .get("page_number")
-                == idx + 1
-                else None
-            )
-            invoice_time = (
-                fields.get("InvoiceTime", default_dict).get("value")
-                if fields.get("InvoiceTime", {})
-                .get("bounding_regions", [{}])[0]
-                .get("page_number")
-                == idx + 1
-                else None
-            )
-            date = combine_date_with_time(date, invoice_time)
-            due_date = (
-                format_date(fields.get("DueDate", default_dict).get("value"))
-                if fields.get("DueDate", {})
-                .get("bounding_regions", [{}])[0]
-                .get("page_number")
-                == idx + 1
-                else None
-            )
-            total_tax = fields.get("TotalTax", {}) or {}
-            total_tax_value = total_tax.get("value", {}) or {}
-            total_tax_bouding_regions = (
-                total_tax.get("bounding_regions", [{}]) or [{}]
-            )[0]
-            taxes = [
-                TaxesInvoice(
-                    value=(
-                        total_tax_value.get("amount")
-                        if total_tax_bouding_regions.get("page_number") == idx + 1
-                        else None
-                    ),
-                    rate=None,
-                )
-            ]
-
-            amount = fields.get("AmountDue", {}) or {}
-            amount_due_value = amount.get("value", {}) or {}
-            amount_due_bouding_regions = (amount.get("bounding_regions", [{}]) or [{}])[
-                0
-            ]
-            amount_due = (
-                amount_due_value.get("amount")
-                if amount_due_bouding_regions.get("page_number") == idx + 1
-                else None
-            )
-            previous_unpaid_balance = (
-                fields.get("PreviousUnpaidBalance", default_dict).get("amount")
-                if fields.get("PreviousUnpaidBalance", {})
-                .get("bounding_regions", [{}])[0]
-                .get("page_number")
-                == idx + 1
-                else None
-            )
+            page_num = idx + 1
+
+            customer_name               = _get_page_val(fields, page_num, ["CustomerName",          "value"          ])
```

Comment:
just out of curiosity, black or flake8 did not reformat your line ? cause it's beautiful like this ahah

Reply:
No, I felt it was more readable to format it that way, with black it becomes a bit cluttered